packages/shared-components/src/components/onto-header/onto-header.tsx

```diff
@@ -1,18 +1,39 @@
 import { Component, Host, h } from '@stencil/core';
+import {WorkbenchServiceProvider, WorkbenchEventService, WorkbenchEventType} from "@ontotext/workbench-api";
 
 @Component({
   tag: 'onto-header',
   styleUrl: 'onto-header.scss',
   shadow: false,
 })
 export class OntoHeader {
+
+  private eventService: WorkbenchEventService;
+
+  private locale = 'en'
+
+  constructor() {
+    this.eventService = WorkbenchServiceProvider.get(WorkbenchEventService);
+  }
+
+  private onClick(): void {
+    // TODO remove this when implement language selector
+    if (this.locale === 'en') {
+      this.locale = 'fr';
+    } else {
+      this.locale = 'en';
+    }
+    this.eventService.emit({NAME: WorkbenchEventType.LANGUAGE_CHANGED, payload: {locale: this.locale}})
```

Comment:
Well. This is another way to emit language chage event. The logic clearly needs to be reconsidered and cleaned up. Right now you have a glovbal context service, a language service and yet here the event is emitted directly via the event service. Why?

Reply:
This is what I wanted to discuss with you and the team. There are two ways for other modules to interact with the global store:

The first is through the global context store using events and subscriptions.
The second is via services. For example, if we need to read or write the selected language, we can interact with the LanguageService.
Depending on the decision, I will rewrite the global and language services.

Reply:
I prefer the service approach. I dont want to drag the entire global store to places where it is not needed, not to mention that it would have a more generic API which might lead to confusion and errors. If I want to subscribe to language change event or emit the same, I'd use the language service and not the generic one. So, in short, stick with the service approach.

Reply:
Refactored.